actors/builtin/miner/sectors.go

```diff
@@ -0,0 +1,89 @@
+package miner
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+
+	"github.com/ipfs/go-cid"
+)
+
+func LoadSectors(store adt.Store, root cid.Cid) (*Sectors, error) {
+	sectorsArr, err := adt.AsArray(store, root)
+	if err != nil {
+		return nil, err
+	}
+	return (*Sectors)(sectorsArr), nil
+}
+
+// Sectors is a helper type for accessing/modifying a miner's sectors. It's safe
+// to pass this object around as needed.
+type Sectors adt.Array
+
+func (sa *Sectors) array() *adt.Array {
+	return (*adt.Array)(sa)
+}
+
+func (sa *Sectors) Root() (cid.Cid, error) {
+	return sa.array().Root()
+}
+
+func (sa *Sectors) LoadSectorInfos(sectorNos *abi.BitField) ([]*SectorOnChainInfo, error) {
+	arr := sa.array()
+
+	var sectorInfos []*SectorOnChainInfo
+	if err := sectorNos.ForEach(func(i uint64) error {
+		var sectorOnChain SectorOnChainInfo
+		found, err := arr.Get(i, &sectorOnChain)
+		if err != nil {
+			return fmt.Errorf("failed to load sector %v: %w", abi.SectorNumber(i), err)
```

Comment:
Use xerrors here and elsewhere.

Reply:
I figured we don't really care about stack traces inside helper types, but that is more consistent.